src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3217,6 +3227,39 @@ Node TheoryStringsRewriter::rewriteStrConvert(Node node)
   return node;
 }
 
+Node TheoryStringsRewriter::rewriteStrReverse(Node node)
+{
+  Assert(node.getKind() == STRING_REV);
+  NodeManager* nm = NodeManager::currentNM();
+  Node x = node[0];
+  if (x.isConst())
+  {
+    std::vector<unsigned> nvec = node[0].getConst<String>().getVec();
+    std::reverse(nvec.begin(), nvec.end());
+    Node retNode = nm->mkConst(String(nvec));
+    return returnRewrite(node, retNode, "str-conv-const");
+  }
+  else if (x.getKind() == STRING_CONCAT)
+  {
+    std::vector<Node> children;
```

Comment:
Could use `NodeBuilder` here.

Reply:
I did not use it since we reverse the child vector below.  Is there a way to reverse the child vector of a NodeBuilder?

Reply:
Oops, you are right of course.